coral-trino/src/main/java/com/linkedin/coral/trino/trino2rel/TrinoConvertletTable.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2018-2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.trino2rel;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.fun.SqlCastFunction;
+import org.apache.calcite.sql2rel.ReflectiveConvertletTable;
+import org.apache.calcite.sql2rel.SqlRexContext;
+import org.apache.calcite.sql2rel.SqlRexConvertlet;
+import org.apache.calcite.sql2rel.StandardConvertletTable;
+
+
+/**
+ * ConvertletTable for Trino Operators
+ * @see ReflectiveConvertletTable documentation for method naming and visibility rules
+ */
+public class TrinoConvertletTable extends ReflectiveConvertletTable {
+
+  @SuppressWarnings("unused")
+  public RexNode convertCast(SqlRexContext cx, SqlCastFunction cast, SqlCall call) {
+    final SqlNode left = call.operand(0);
+    RexNode leftRex = cx.convertExpression(left);
+    SqlDataTypeSpec dataType = call.operand(1);
+    RelDataType castType = dataType.deriveType(cx.getValidator(), true);
+    // can not call RexBuilder.makeCast() since that optimizes to remove the cast
+    // we don't want to remove the cast
+    return cx.getRexBuilder().makeAbstractCast(castType, leftRex);
+  }
```

Comment:
As of now, we still need this. This step happens during the SqlNode to RelNode phase meaning that not all the conversion happens during the SqlNode layer. Maybe this could be something to work on in the future.